packages/react-components/react-table/src/hooks/useSelection.ts

```diff
@@ -1,14 +1,38 @@
-import { GetRowIdInternal, SelectionStateInternal, UseTableOptions } from './types';
-import { useMultipleSelection } from './useMultipleSelection';
-import { useSingleSelection } from './useSingleSelection';
+import * as React from 'react';
+import { useEventCallback, usePrevious } from '@fluentui/react-utilities';
+import { createSelectionManager } from './selectionManager';
+import { GetRowIdInternal, RowId, SelectionMode, SelectionStateInternal } from './types';
 
 export function useSelection<TItem>(
-  selectionMode: UseTableOptions<TItem>['selectionMode'],
+  selectionMode: SelectionMode,
```

Comment:
I still believe having a hook for each selection mode is better than having a single one that internalizes both. It would be less of an impact on the bundle size (since only one of the selection managers would be imported) and IMO more transparent as you'd have at the name of the function it's actual behaviour

Reply:
I still have a plan to pull out the hooks completely from `useTable` but while they are built in there's no sense to do that hook separation now